pkg/aws/g8scontrolplane/admit_g8scontrolplane.go

```diff
@@ -113,10 +118,15 @@ func (a *Admitter) Admit(request *v1beta1.AdmissionRequest) ([]admission.PatchOp
 			}
 			// If there is an AWSControlPlane, the default replicas match the number of AZs
 			replicas = len(awsControlPlane.Spec.AvailabilityZones)
+			// If there is an AWSControlplane, we get its infrastructure reference
+			infrastructureCRRef, err = reference.GetReference(infrastructurev1alpha2scheme.Scheme, awsControlPlane)
+			if err != nil {
+				return microerror.Maskf(aws.ExecutionFailedError, "failed to create reference to AWSControlplane: %v", err)
+			}
 
 			// If the availability zones need to be updated from 1 to 3, we do it here
 			{
-				if g8sControlPlaneNewCR.Spec.Replicas == 3 && g8sControlPlaneOldCR.Spec.Replicas == 1 && len(awsControlPlane.Spec.AvailabilityZones) == 1 {
+				if aws.IsHAVersion(releaseVersion) && g8sControlPlaneNewCR.Spec.Replicas == 3 && g8sControlPlaneOldCR.Spec.Replicas == 1 && len(awsControlPlane.Spec.AvailabilityZones) == 1 {
```

Comment:
These conditions do not get better for humans. As an alternative you could define a helper function or at least variable that has a more telling name to it. Just to reduce complexity at a given moment.